packages/vite-plugin-svelte/src/ui/inspector/plugin.ts

```diff
@@ -0,0 +1,98 @@
+import { createRequire } from 'module';
+import { Plugin } from 'vite';
+import { log } from '../../utils/log';
+import { InspectorOptions } from '../../utils/options';
+
+const defaultInspectorOptions: InspectorOptions = {
+	holdKey: 's',
+	toggleKeyCombo: 'alt-s',
+	showToggleButton: true,
+	customStyles: true
+};
+
+export function svelteInspector(): Plugin {
+	let require: NodeRequire;
+	let inspectorOptions: InspectorOptions;
+	let append_to: string | undefined;
+
+	return {
+		name: 'vite-plugin-svelte:inspector',
+		apply: 'serve',
+		enforce: 'pre',
+
+		configResolved(config) {
+			const vps = config.plugins.find((p) => p.name === 'vite-plugin-svelte');
+			if (vps?.api?.options?.experimental?.inspector) {
+				inspectorOptions = {
+					...defaultInspectorOptions,
+					...vps.api.options.experimental.inspector
+				};
+			}
+			if (!vps || !inspectorOptions) {
+				// disabled, turn all hooks into noops
+				this.resolveId = this.load = this.transformIndexHtml = this.transform = () => {};
+			} else {
+				require = createRequire(config.root || process.cwd());
+				if (vps.api.options.kit && !inspectorOptions.appendTo) {
+					const out_dir = vps.api.options.kit.outDir || '.svelte-kit';
+					inspectorOptions.appendTo = `${out_dir}/runtime/client/start.js`;
```

Comment:
as discussed this is a bit brittle, ideas how to load the inspector when transformIndexHtml isn't there are welcome.

At least we are in control of appendTo and the runtime/client/start.js bit